### PR TITLE
Bump version for  TSP and TCGC

### PR DESCRIPTION
**Pull request by live1206**
Fix https://github.com/microsoft/typespec/issues/3891

### CODE REVIEW

**Comment by live1206** [comment]
Regen in https://github.com/Azure/autorest.csharp/pull/4982

File: packages/http-client-csharp/emitter/src/lib/client-model-builder.ts [hunk 1]
```diff
@@ -126,43 +127,26 @@ export function createModel(sdkContext: SdkContext<NetEmitterOptions>): CodeMode
...
-      ];
+    // TODO: handle SdkUnionType
+    if (p.type.kind === "union") {
+      return fromSdkEndpointType(p.type.values[0] as SdkEndpointType);
+    } else {
+      return fromSdkEndpointType(p.type);
+    }
+  }
 
+  function fromSdkEndpointType(type: SdkEndpointType): InputParameter[] {
```
Comment (ArcturusZhang) [inline packages/http-client-csharp/emitter/src/lib/client-model-builder.ts:138]: we have a `throw` branch in `converter.ts` for Endpoint type.
Could we consolidate this method with that?

Comment (ArcturusZhang) [reply packages/http-client-csharp/emitter/src/lib/client-model-builder.ts:138]: or at least we should add a todo here.

File: packages/http-client-csharp/eng/scripts/Generate.ps1 [hunk 2]
```diff
@@ -16,6 +16,7 @@ if ($null -eq $filter -or $filter -eq "Unbranded-TypeSpec") {
     $testProjectsLocalDir = Join-Path $packageRoot 'generator' 'TestProjects' 'Local'
 
     $unbrandedTypespecTestProject = Join-Path $testProjectsLocalDir "Unbranded-TypeSpec"
+    $unbrandedTypespecTestProject = $unbrandedTypespecTestProject.Replace("\", "/")  # replace \ with / for the path to avoid path-unix-style warning
```
Comment (archerzz) [inline packages/http-client-csharp/eng/scripts/Generate.ps1:19]: There are too many `"\"` to `"/"` conversion, maybe we can use `slash` module: https://www.npmjs.com/package/slash

**Review by archerzz — APPROVED** [review]

File: packages/http-client-csharp/eng/scripts/Generate.ps1 (hunk 2 above)
Comment (live1206) [reply packages/http-client-csharp/eng/scripts/Generate.ps1:19]: hmm, this is `Powershell` not `js` though.

File: packages/http-client-csharp/emitter/src/lib/client-model-builder.ts (hunk 1 above)
Comment (live1206) [reply packages/http-client-csharp/emitter/src/lib/client-model-builder.ts:137]: consolidated, we should throw for endpoint in `converter.ts` as well.

**Review by ArcturusZhang — APPROVED** [review]

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**